packages/b/bullet3/xmake.lua

```diff
@@ -3,19 +3,54 @@ package("bullet3")
     set_homepage("http://bulletphysics.org")
     set_description("Bullet Physics SDK.")
 
-    set_urls("https://github.com/bulletphysics/bullet3/archive/$(version).zip",
-             "https://github.com/bulletphysics/bullet3.git")
-    add_versions("2.88", "f361d10961021a186b80821cfc1cfafc8dac48ce35f7d5e8de0943af4b3ddce4")
+    add_urls("https://github.com/bulletphysics/bullet3/archive/$(version).zip", {alias = "zip"})
+    add_urls("https://github.com/bulletphysics/bullet3.git", {alias = "git"})
+    add_versions("zip:2.88", "f361d10961021a186b80821cfc1cfafc8dac48ce35f7d5e8de0943af4b3ddce4")
+    add_versions("git:2.88", "126b6762200b1a9ec8df39b8fd53cb2ad62fb146")
+    add_versions("zip:3.05", "e7ef322d8038e397cd6d79145a856cf5b4d558ce091d49b5239d625a46fef0d7")
+    add_versions("git:3.05", "b12da620abdff665d53cd953118a741758f28523")
```

Comment:
2.88/3.05 is the tag/release of bullet3. If it is not a special case, it will generally not change.

We need only write:

```lua
    set_urls("https://github.com/bulletphysics/bullet3/archive/$(version).zip",
             "https://github.com/bulletphysics/bullet3.git")
    add_versions("3.05", "e7ef322d8038e397cd6d79145a856cf5b4d558ce091d49b5239d625a46fef0d7")
    add_versions("2.88", "f361d10961021a186b80821cfc1cfafc8dac48ce35f7d5e8de0943af4b3ddce4")

```

Reply:
This is how it was before I changed it and xmake tried to checkout the checksum and failed.

Reply:
```
downloading bullet3 .. (1/git) error: @programdir\core\sandbox\modules\os.lua:258: error: pathspec 'e7ef322d8038e397cd6d79145a856cf5b4d558ce091d49b5239d625a46fef0d7' did not match any file(s) known to git
```

Reply:
It's bug, I have fixed it on dev branch. and you can try it again.

Reply:
I updated using ```xmake update dev```. My version is now ```v2.3.7+202010050918```. But I still have the same bug.

Reply:
win?

Reply:
Windows 10

Reply:
appveyor ci has not packaged the installer yet for windows, you need to wait a few minutes and update it.

Or you can download install package form https://github.com/xmake-io/xmake/actions/runs/289413520